File: csharp/robot-name/RobotName.cs
using System;

public class Robot
{
  public string Name { get; private set; }

  public Robot()
  {
    Reset();
  }

  public void Reset()
  {
    Name = RobotName();
  }

  private string RobotName()
  {
    return GetLetter() + GetLetter() + GetNumbers();
  }

  private static Random random = new Random();

  private string GetLetter()
  {
    return ((char)('A' + random.Next(26))).ToString();
  }

  private string GetNumbers()
  {
    return (random.Next(1000)).ToString("000"); 
  }
}

File: javascript/robot-name/robot-name.js
var letters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ",
    bots = {};

function getRandom(max) {
  return parseInt(Math.random() * max);
};

function createName() {
  var name = letters[getRandom(26)] + letters[getRandom(26)]
             + Math.random().toString().substr(-3);

  if (bots[name]) return createName();
  else bots[name] = true;
  return name;
};

module.exports = function() {
  this.name = createName();
  this.reset = function() {
    this.name = createName();
  };
};
